app/grandchallenge/evaluation/admin.py

```diff
@@ -58,6 +65,84 @@ def clean(self):
         return cleaned_data
 
 
+class ConfigureAlgorithmPhasesForm(Form):
+    phases = ModelMultipleChoiceField(
+        queryset=Phase.objects.select_related("challenge")
+        .filter(submission_kind=SubmissionKindChoices.CSV)
+        .all()
+    )
+    algorithm_inputs = ModelMultipleChoiceField(
+        queryset=ComponentInterface.objects.all()
+    )
+    algorithm_outputs = ModelMultipleChoiceField(
+        queryset=ComponentInterface.objects.all()
+    )
+
+    def clean(self):
+        cleaned_data = super().clean()
+        try:
+            for phase in self.cleaned_data["phases"]:
+                if Archive.objects.filter(
+                    slug=f"{slugify(phase.challenge.short_name)}-{slugify(phase.title)}-dataset"
+                ).exists():
+                    raise ValidationError(
+                        f"Archive for {phase} already exists."
+                    )
+        except KeyError:
+            pass
+        return cleaned_data
+
+
+class ConfigureAlgorithmPhasesView(PermissionRequiredMixin, FormView):
+    form_class = ConfigureAlgorithmPhasesForm
+    permission_required = "evaluation.configure_algorithm_phase"
+    template_name = "admin/evaluation/phase/algorithm_phase_form.html"
+    raise_exception = True
+
+    def form_valid(self, form):
+        for phase in form.cleaned_data["phases"]:
+            self.turn_phase_into_algorithm_phase(
+                phase=phase,
+                inputs=form.cleaned_data["algorithm_inputs"],
+                outputs=form.cleaned_data["algorithm_outputs"],
+            )
+        messages.success(self.request, "Phases were successfully updated")
+        return super().form_valid(form)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        context.update(
+            {
+                **admin.site.each_context(self.request),
+                "opts": Phase._meta,
+                "title": "Turn phase(s) into algorithm phase(s)",
```

Comment:
There is a lot going on in this admin integration that I do not really understand and may be fragile. Would it be more straightforward using a normal view? Especially now you have the specialised permission.

Reply:
I can turn it into a normal view, sure. The extra context here is so that the sidebar and the breadcrumbs get displayed.

Reply:
I think a normal view is best for this. The very first version of GC had lots of custom modifications of the admin which were tedious to maintain long term, so I ended up rewriting everything as normal views.

Reply:
Ok, understood. Unsure now about the permissions: the specific new permission I added is enough if you only grant it to staff users, but normally admin views are locked down to staff users additionally. Should I add a check for that as well, just in case and to be in line with other admin views?

Reply:
Staff should be limited to checking for the Django admin access. Having a global permission check on a user for a normal view should be sufficient, right? It's the same as what we have for say, blog post creation. We could eventually create a group of normal users (like the DIAG 10% team) who can onboard challenges.